packages/node/src/indexer/fetch.service.ts

```diff
@@ -481,23 +502,56 @@ export class FetchService implements OnApplicationShutdown {
           this.eventEmitter.emit(IndexerEvent.SkipDictionary);
         }
       }
-
       const endHeight = this.nextEndBlockHeight(
         startBlockHeight,
         scaledBatchSize,
       );
 
       if (handlers.length && this.getModulos().length === handlers.length) {
-        this.blockDispatcher.enqueueBlocks(
-          this.getEnqueuedModuloBlocks(startBlockHeight),
-        );
+        const enqueuingBlocks = this.getEnqueuedModuloBlocks(startBlockHeight);
+        const cleanedBatchBlocks = this.filteredBlockBatch(enqueuingBlocks);
+        this.blockDispatcher.enqueueBlocks(cleanedBatchBlocks);
+        this.setLatestBufferHeight(cleanedBatchBlocks, enqueuingBlocks);
       } else {
-        this.blockDispatcher.enqueueBlocks(
-          range(startBlockHeight, endHeight + 1),
-        );
+        const enqueuingBlocks = range(startBlockHeight, endHeight + 1);
+        const cleanedBatchBlocks = this.filteredBlockBatch(enqueuingBlocks);
+        this.blockDispatcher.enqueueBlocks(cleanedBatchBlocks);
+        this.setLatestBufferHeight(cleanedBatchBlocks, enqueuingBlocks);
```

Comment:
Maybe `enqueueBlocks` should take a `lastBufferedHeight` parameter as we always call the 2 together

Reply:
Maybe we can set `enqueueBlocks(cleanedBatchBlocks: number[],  lastBufferedHeight?: number )`. 
If `lastBufferedHeight` is defined, set it to blockDispatcher, otherwise get last one from `cleanedBatchBlocks `